<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>ftir-uq.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown #bookdown:version# and GitBook 2.6.7" />

  <meta property="og:title" content="ftir-uq.knit" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="ftir-uq" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="ftir-uq.knit" />
  
  
  

<meta name="author" content="Christian Krohn, PhD, RMIT University" />


<meta name="date" content="2024-10-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<!--bookdown:link_prev-->
<!--bookdown:link_next-->
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



<!--bookdown:title:start-->
<!--bookdown:title:end-->

<!--bookdown:toc:start-->
  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">
<!--bookdown:toc2:start-->
<ul>
<li><a href="#ftir-data-assessments-in-collaboration-with-uq" id="toc-ftir-data-assessments-in-collaboration-with-uq"><span class="toc-section-number">1</span> FTIR data assessments in collaboration with UQ</a>
<ul>
<li><a href="#import" id="toc-import"><span class="toc-section-number">1.1</span> Import</a></li>
<li><a href="#microbial-16s-v3v4-pca" id="toc-microbial-16s-v3v4-pca"><span class="toc-section-number">1.2</span> Microbial 16S (V3V4) PCA</a></li>
<li><a href="#raw-spectra" id="toc-raw-spectra"><span class="toc-section-number">1.3</span> Raw spectra</a>
<ul>
<li><a href="#pca-atr" id="toc-pca-atr"><span class="toc-section-number">1.3.1</span> PCA ATR</a></li>
<li><a href="#pca-ave" id="toc-pca-ave"><span class="toc-section-number">1.3.2</span> PCA AVE</a></li>
<li><a href="#comparing-atr-and-ave-of-pc3" id="toc-comparing-atr-and-ave-of-pc3"><span class="toc-section-number">1.3.3</span> Comparing ATR and AVE of PC3</a></li>
<li><a href="#spectra-contributions-to-pc3" id="toc-spectra-contributions-to-pc3"><span class="toc-section-number">1.3.4</span> Spectra contributions to PC3</a></li>
</ul></li>
<li><a href="#transformed-spectra---centred-log-ratio" id="toc-transformed-spectra---centred-log-ratio"><span class="toc-section-number">1.4</span> Transformed spectra - centred-log ratio</a>
<ul>
<li><a href="#pca-atr-clr" id="toc-pca-atr-clr"><span class="toc-section-number">1.4.1</span> PCA ATR clr</a></li>
<li><a href="#pca-ave---clr" id="toc-pca-ave---clr"><span class="toc-section-number">1.4.2</span> PCA AVE - clr</a></li>
<li><a href="#comparing-atr-and-ave-of-pc3-1" id="toc-comparing-atr-and-ave-of-pc3-1"><span class="toc-section-number">1.4.3</span> Comparing ATR and AVE of PC3</a></li>
<li><a href="#spectra-contributions-to-pc3-1" id="toc-spectra-contributions-to-pc3-1"><span class="toc-section-number">1.4.4</span> Spectra contributions to PC3</a></li>
</ul></li>
<li><a href="#corplot" id="toc-corplot"><span class="toc-section-number">1.5</span> Corplot</a></li>
</ul></li>
</ul>
<!--bookdown:toc2:end-->
      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<!--bookdown:toc:end-->
<!--bookdown:body:start-->
<div id="ftir-data-assessments-in-collaboration-with-uq" class="section level1 hasAnchor" number="1">
<h1 class="hasAnchor"><span class="header-section-number">1</span> FTIR data assessments in collaboration with UQ<a href="#ftir-data-assessments-in-collaboration-with-uq" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="import" class="section level2 hasAnchor" number="1.1">
<h2 class="hasAnchor"><span class="header-section-number">1.1</span> Import<a href="#import" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># PREPARE</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(ggpubr)       <span class="co"># a handy helper package for ggplots</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())  <span class="co"># setting the theme for ggplots</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">library</span>(microbiome)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="st">`</span><span class="at">%notin%</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">Negate</span>(<span class="st">`</span><span class="at">%in%</span><span class="st">`</span>)</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="co"># MICROBIAL DATA</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="co"># reading in a previously saved phyloseq object. This contains microbial 16S-ASV abundances of all samples. </span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>ps_1C <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/ps_215samplesJul24&quot;</span>)</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="co"># ps_1C</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a><span class="co">#phyloseq-class experiment-level object</span></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a><span class="co">#otu_table()   OTU Table:         [ 17961 taxa and 215 samples ]</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="co">#sample_data() Sample Data:       [ 215 samples by 48 sample variables ]</span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a><span class="co">#tax_table()   Taxonomy Table:    [ 17961 taxa by 7 taxonomic ranks ]</span></span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a><span class="co">#phy_tree()    Phylogenetic Tree: [ 17961 tips and 17822 internal nodes ]</span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a><span class="do">### Pre-filtering microbial data</span></span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a>ps.flt  <span class="ot">&lt;-</span>  <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(ps_1C) <span class="sc">&gt;=</span> <span class="dv">10</span>, ps_1C) <span class="co">#m inimum reads per ASV</span></span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a><span class="co">#phyloseq-class experiment-level object</span></span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a><span class="co">#otu_table()   OTU Table:         [ 5855 taxa and 215 samples ]</span></span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a><span class="co">#sample_data() Sample Data:       [ 215 samples by 60 sample variables ]</span></span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a><span class="co">#tax_table()   Taxonomy Table:    [ 5855 taxa by 7 taxonomic ranks ]</span></span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a><span class="co">#phy_tree()    Phylogenetic Tree: [ 5855 tips and 5763 internal nodes ]</span></span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a></span>
<span id="cb1-28"><a href="#cb1-28" tabindex="-1"></a><span class="co"># MASTERDATA</span></span>
<span id="cb1-29"><a href="#cb1-29" tabindex="-1"></a><span class="co"># change this path to where you store the .csv file. you can provide an absolute path like this. Check the path syntax for Windows as it might be different. </span></span>
<span id="cb1-30"><a href="#cb1-30" tabindex="-1"></a>masterdata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;./data/masterdataProject1C_May24.csv&quot;</span>)</span>
<span id="cb1-31"><a href="#cb1-31" tabindex="-1"></a><span class="do">## Change date to date format in R </span></span>
<span id="cb1-32"><a href="#cb1-32" tabindex="-1"></a>masterdata <span class="ot">&lt;-</span>  masterdata <span class="sc">%&gt;%</span></span>
<span id="cb1-33"><a href="#cb1-33" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Date =</span> lubridate<span class="sc">::</span><span class="fu">dmy</span>(<span class="fu">as.character</span>(Date)) )</span>
<span id="cb1-34"><a href="#cb1-34" tabindex="-1"></a><span class="co"># Make AD, Treatment and SludgeType = factors</span></span>
<span id="cb1-35"><a href="#cb1-35" tabindex="-1"></a>masterdata<span class="sc">$</span>AD <span class="ot">&lt;-</span> <span class="fu">factor</span>(masterdata<span class="sc">$</span>AD, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;AD1&quot;</span>, <span class="st">&quot;AD2&quot;</span>,<span class="st">&quot;AD3&quot;</span>, <span class="st">&quot;AD4&quot;</span>,<span class="st">&quot;AD5&quot;</span>, <span class="st">&quot;AD6&quot;</span>, <span class="st">&quot;Full-scale&quot;</span>, <span class="st">&quot;PSTWAS&quot;</span>))</span>
<span id="cb1-36"><a href="#cb1-36" tabindex="-1"></a>masterdata<span class="sc">$</span>SludgeType <span class="ot">&lt;-</span> <span class="fu">factor</span>(masterdata<span class="sc">$</span>SludgeType, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Control&quot;</span>, <span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;Foam&quot;</span>, <span class="st">&quot;Full-scale&quot;</span>, <span class="st">&quot;PSTWAS&quot;</span>))</span>
<span id="cb1-37"><a href="#cb1-37" tabindex="-1"></a>masterdata<span class="sc">$</span>Treatment <span class="ot">&lt;-</span> <span class="fu">factor</span>(masterdata<span class="sc">$</span>Treatment, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Control&quot;</span>, <span class="st">&quot;Treatment&quot;</span>, <span class="st">&quot;Full-scale&quot;</span>, <span class="st">&quot;PSTWAS&quot;</span>))</span>
<span id="cb1-38"><a href="#cb1-38" tabindex="-1"></a>masterdata<span class="sc">$</span>Period <span class="ot">&lt;-</span> <span class="fu">factor</span>(masterdata<span class="sc">$</span>Period, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Converging&quot;</span>, <span class="st">&quot;SteadyState&quot;</span>, <span class="st">&quot;Glycerol&quot;</span>, <span class="st">&quot;Inhibition&quot;</span>, <span class="st">&quot;Recovery/Feedingpause&quot;</span>, <span class="st">&quot;Recovery/Foaming&quot;</span>, <span class="st">&quot;Recovery/Postfoam&quot;</span>, <span class="st">&quot;SteadyState/Postfoam&quot;</span>))</span>
<span id="cb1-39"><a href="#cb1-39" tabindex="-1"></a></span>
<span id="cb1-40"><a href="#cb1-40" tabindex="-1"></a></span>
<span id="cb1-41"><a href="#cb1-41" tabindex="-1"></a><span class="co"># LOAD REFLECTANCE DATA </span></span>
<span id="cb1-42"><a href="#cb1-42" tabindex="-1"></a>ATR <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;./data/ADVATR.csv&quot;</span>)[<span class="sc">-</span><span class="dv">1</span>,] </span>
<span id="cb1-43"><a href="#cb1-43" tabindex="-1"></a>ATR <span class="ot">&lt;-</span> ATR <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>)<span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">&quot;X&quot;</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>ID) <span class="sc">%&gt;%</span> </span>
<span id="cb1-44"><a href="#cb1-44" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(S537_AD1<span class="sc">:</span>S1370_AD6, as.numeric))</span>
<span id="cb1-45"><a href="#cb1-45" tabindex="-1"></a><span class="co"># OPTION 1: raw</span></span>
<span id="cb1-46"><a href="#cb1-46" tabindex="-1"></a>ATR <span class="ot">&lt;-</span>  <span class="fu">as.matrix</span>(ATR)</span>
<span id="cb1-47"><a href="#cb1-47" tabindex="-1"></a><span class="fu">head</span>(ATR) <span class="sc">%&gt;%</span> <span class="fu">datatable</span>(<span class="at">caption =</span> <span class="st">&quot;ATR raw&quot;</span>) </span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-a17945d6b6d09332ed64" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-a17945d6b6d09332ed64">{"x":{"filter":"none","vertical":false,"caption":"<caption>ATR raw<\/caption>","data":[["400.1635","400.6456","401.1277","401.6098","402.0919","402.5741"],[0.00974971,0.01666979,0.01778047,0.02032463,0.02102293,0.02261765],[0.007554614,0.01249666,0.01374776,0.01641333,0.0172995,0.01918816],[0.01179065,0.01794195,0.01864124,0.02072328,0.02113081,0.02234364],[0.007212107,0.01243628,0.01358758,0.01622816,0.01705122,0.0189135],[0.01229677,0.01830757,0.01896677,0.02088482,0.02128498,0.02247025],[0.009500721,0.01597138,0.01694473,0.01936648,0.0200798,0.02167596],[0.009964000000000001,0.0166565,0.01751959,0.01997018,0.02055308,0.02209297],[0.01017251,0.01627371,0.01715273,0.01929551,0.01988415,0.021235],[0.009912694999999999,0.01651472,0.01743678,0.01978466,0.02036035,0.02182387],[0.006387523,0.01116406,0.01251171,0.01476368,0.01563543,0.01713906],[0.007334448,0.01300852,0.01433291,0.016818,0.01766051,0.01928827],[0.008678745,0.01499203,0.01599591,0.0184179,0.01909436,0.02063176],[0.01232022,0.01842999,0.01921379,0.02128869,0.02178312,0.02302029],[0.007937674,0.0141537,0.01551829,0.01824683,0.01912828,0.02088464],[0.01129173,0.01763442,0.01845169,0.02066719,0.02119034,0.02252601],[0.008998925,0.01584287,0.01711947,0.01979976,0.02057392,0.02220075],[-0.008070819,0.005650214,0.01408485,0.01250048,0.0177475,0.01655202],[0.008092837,0.01436133,0.0156611,0.01823142,0.01904364,0.0206647],[0.009342408999999999,0.01590983,0.01684521,0.01926081,0.019845,0.021339],[0.008255261,0.01471787,0.01600412,0.01868906,0.01948524,0.02119628],[0.008953509,0.01574798,0.01704962,0.01965586,0.02039491,0.02198978],[-0.008121588000000001,0.005757887,0.01424501,0.01261263,0.01780619,0.01657043],[0.009004791,0.0156921,0.01674284,0.01926626,0.01987535,0.02137785],[0.007258048,0.01260529,0.01383817,0.01646068,0.01728185,0.01905566]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>S537_AD1<\/th>\n      <th>S538_AD2<\/th>\n      <th>S539_AD3<\/th>\n      <th>S540_AD4<\/th>\n      <th>S541_AD5<\/th>\n      <th>S542_AD6<\/th>\n      <th>S958_AD1<\/th>\n      <th>S959_AD2<\/th>\n      <th>S960_AD3<\/th>\n      <th>S961_AD4<\/th>\n      <th>S962_AD5<\/th>\n      <th>S963_AD6<\/th>\n      <th>S1189_AD1<\/th>\n      <th>S1190_AD2<\/th>\n      <th>S1191_AD3<\/th>\n      <th>S1192_AD4<\/th>\n      <th>S1193_AD5<\/th>\n      <th>S1194_AD6<\/th>\n      <th>S1365_AD1<\/th>\n      <th>S1366_AD2<\/th>\n      <th>S1367_AD3<\/th>\n      <th>S1368_AD4<\/th>\n      <th>S1369_AD5<\/th>\n      <th>S1370_AD6<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"S537_AD1","targets":1},{"name":"S538_AD2","targets":2},{"name":"S539_AD3","targets":3},{"name":"S540_AD4","targets":4},{"name":"S541_AD5","targets":5},{"name":"S542_AD6","targets":6},{"name":"S958_AD1","targets":7},{"name":"S959_AD2","targets":8},{"name":"S960_AD3","targets":9},{"name":"S961_AD4","targets":10},{"name":"S962_AD5","targets":11},{"name":"S963_AD6","targets":12},{"name":"S1189_AD1","targets":13},{"name":"S1190_AD2","targets":14},{"name":"S1191_AD3","targets":15},{"name":"S1192_AD4","targets":16},{"name":"S1193_AD5","targets":17},{"name":"S1194_AD6","targets":18},{"name":"S1365_AD1","targets":19},{"name":"S1366_AD2","targets":20},{"name":"S1367_AD3","targets":21},{"name":"S1368_AD4","targets":22},{"name":"S1369_AD5","targets":23},{"name":"S1370_AD6","targets":24}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># OPTION 2: clr transformed</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>ATRclr <span class="ot">&lt;-</span> compositions<span class="sc">::</span><span class="fu">clr</span>(ATR)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">head</span>(ATRclr) <span class="sc">%&gt;%</span> <span class="fu">datatable</span>(<span class="at">caption =</span> <span class="st">&quot;ATR clr&quot;</span>) </span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-7c26f4a660c6835a887e" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-7c26f4a660c6835a887e">{"x":{"filter":"none","vertical":false,"caption":"<caption>ATR clr<\/caption>","data":[["400.1635","400.6456","401.1277","401.6098","402.0919","402.5741"],[0.07561012945734547,0.1735699784413942,0.09827075793462559,0.1134511767715676,0.08468972014984155,0.09302882462941309],[-0.1794689090929138,-0.1145667121574805,-0.1589540042469375,-0.1002884647533655,-0.1102464272891042,-0.0714083281490967],[0.265679432976623,0.2471134256144243,0.1455464246525171,0.1328754317067107,0.08980813837426282,0.0808399665061672],[-0.2258662699055005,-0.1194101135401882,-0.1706737658029969,-0.1116342695168751,-0.1247022712192281,-0.08582579749981889],[0.3077092147461933,0.2672865147538976,0.1628585946065106,0.1406403049716234,0.09707763300975758,0.0864904617819775],[0.04974027894316002,0.1307702497547041,0.05012696503078118,0.06516146227452113,0.03879030797349303,0.05050206204417007],[0.09735118587961633,0.172772409872886,0.08348977741045349,0.09585788659281613,0.0620867814160353,0.06955770807273032],[0.1180615724154235,0.1495228015189145,0.06232743861956092,0.06149015204376518,0.02899890531667415,0.02994901346616174],[0.09218884737830191,0.1642239836535877,0.07875186240905396,0.08652461563925051,0.05266535546324036,0.05730257459936938],[-0.3472808552141986,-0.2273284306840502,-0.2531649002124947,-0.206212163724703,-0.2113845337109841,-0.1843416811070879],[-0.2090452583809412,-0.07442359335524174,-0.1172716142919121,-0.07593253290386048,-0.08959295306004433,-0.06620461117724119],[-0.04075047855497083,0.06749060579984434,-0.007496841435338908,0.01494074349869967,-0.01153102315267418,0.001129885796649077],[0.3096144035684762,0.273951108326532,0.1757983443997873,0.1597936714623756,0.1202113312930497,0.1106742506564076],[-0.1300071262819325,0.009947953245722729,-0.03781057772508678,0.005609091996442217,-0.009756158365500589,0.01331221024988682],[0.2224431878517965,0.2298245532447876,0.1353260592101195,0.130165144322449,0.09262139081480214,0.08896889125779062],[-0.004522285762599054,0.1226914360811531,0.06038650629298292,0.08728754210033918,0.06310022765736978,0.07442432179228575],[0,-0.9083347002141924,-0.1347301543531163,-0.3726152307435724,-0.08467936570966605,-0.219193601543366],[-0.1106480620720562,0.02451105695420619,-0.02864997524527446,0.004764204924465609,-0.01419083907915519,0.002725180259463666],[0.03293673037624245,0.1269090364274454,0.04423643739062921,0.05969018724584707,0.02702805959395871,0.03483463412041532],[-0.09077671744634408,0.04903428098210583,-0.00698371690513433,0.02955605161309194,0.008733229240798668,0.02812394428382126],[-0.009581889032646451,0.1166839823166894,0.05629801009364765,0.07999323837182359,0.05436133327242931,0.0648760496700298],[0,-0.8894575536337879,-0.1234232360471541,-0.3636835814530723,-0.08137787698603516,-0.2180819686843343],[-0.003870642489889775,0.1131292802395194,0.03814079819880867,0.05997310520645804,0.02855624380716337,0.03665358896898585],[-0.2195164893591883,-0.1059115536428674,-0.1523991899840373,-0.09740776764679637,-0.1112672088105078,-0.07833757999475566]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>S537_AD1<\/th>\n      <th>S538_AD2<\/th>\n      <th>S539_AD3<\/th>\n      <th>S540_AD4<\/th>\n      <th>S541_AD5<\/th>\n      <th>S542_AD6<\/th>\n      <th>S958_AD1<\/th>\n      <th>S959_AD2<\/th>\n      <th>S960_AD3<\/th>\n      <th>S961_AD4<\/th>\n      <th>S962_AD5<\/th>\n      <th>S963_AD6<\/th>\n      <th>S1189_AD1<\/th>\n      <th>S1190_AD2<\/th>\n      <th>S1191_AD3<\/th>\n      <th>S1192_AD4<\/th>\n      <th>S1193_AD5<\/th>\n      <th>S1194_AD6<\/th>\n      <th>S1365_AD1<\/th>\n      <th>S1366_AD2<\/th>\n      <th>S1367_AD3<\/th>\n      <th>S1368_AD4<\/th>\n      <th>S1369_AD5<\/th>\n      <th>S1370_AD6<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"S537_AD1","targets":1},{"name":"S538_AD2","targets":2},{"name":"S539_AD3","targets":3},{"name":"S540_AD4","targets":4},{"name":"S541_AD5","targets":5},{"name":"S542_AD6","targets":6},{"name":"S958_AD1","targets":7},{"name":"S959_AD2","targets":8},{"name":"S960_AD3","targets":9},{"name":"S961_AD4","targets":10},{"name":"S962_AD5","targets":11},{"name":"S963_AD6","targets":12},{"name":"S1189_AD1","targets":13},{"name":"S1190_AD2","targets":14},{"name":"S1191_AD3","targets":15},{"name":"S1192_AD4","targets":16},{"name":"S1193_AD5","targets":17},{"name":"S1194_AD6","targets":18},{"name":"S1365_AD1","targets":19},{"name":"S1366_AD2","targets":20},{"name":"S1367_AD3","targets":21},{"name":"S1368_AD4","targets":22},{"name":"S1369_AD5","targets":23},{"name":"S1370_AD6","targets":24}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># CREATE METADATA FOR 24 SELECTED SAMPLES AND MATCH SAMPLE NAMES WITH SPECTRA DF</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>filtervec <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;2023-05-01&quot;</span>, <span class="st">&quot;2023-05-15&quot;</span>, <span class="st">&quot;2023-06-07&quot;</span>,<span class="st">&quot;2023-06-12&quot;</span>)  <span class="co"># 12thJuneDNA = 13th June EPS</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>filtervec2 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;AD1&quot;</span>, <span class="st">&quot;AD2&quot;</span>, <span class="st">&quot;AD3&quot;</span>, <span class="st">&quot;AD4&quot;</span>, <span class="st">&quot;AD5&quot;</span>,<span class="st">&quot;AD6&quot;</span>)  <span class="co"># check</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> masterdata <span class="sc">%&gt;%</span> </span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Date <span class="sc">%in%</span> filtervec <span class="sc">&amp;</span> </span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>                  AD <span class="sc">%in%</span> filtervec2)</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>metadata<span class="sc">$</span>SampleID.EPS <span class="ot">&lt;-</span> <span class="fu">colnames</span>(ATR)</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">&quot;SampleID.EPS&quot;</span>)</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="fu">head</span>(metadata ) <span class="sc">%&gt;%</span> <span class="fu">datatable</span>(<span class="at">caption =</span> <span class="st">&quot;metadata for 24 selected samples&quot;</span>) </span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-49bd83d123490135a9d9" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-49bd83d123490135a9d9">{"x":{"filter":"none","vertical":false,"caption":"<caption>metadata for 24 selected samples<\/caption>","data":[["S537_AD1","S538_AD2","S539_AD3","S540_AD4","S541_AD5","S542_AD6"],[359,360,361,362,363,364],["2023-05-01","2023-05-01","2023-05-01","2023-05-01","2023-05-01","2023-05-01"],["AD1","AD2","AD3","AD4","AD5","AD6"],["Control","Treatment","Control","Control","Treatment","Treatment"],["Control","Treatment","Control","Control","Treatment","Treatment"],[61,61,61,61,61,61],[10,10,10,10,10,10],["SteadyState","SteadyState","SteadyState","SteadyState","SteadyState","SteadyState"],["#048","#049","#050","#051","#052","#053"],["","","","","",""],[null,null,null,null,null,null],[null,null,null,null,null,null],[null,null,null,null,null,null],[null,null,null,null,null,null],["","","","","",""],[null,null,null,null,null,null],["S525_AD1","S526_AD2","S527_AD3","S528_AD4","S529_AD5","S530_AD6"],[21952,21487,22184.5,20276,20644,20324.5],[1000,1100,1162.5,1162.5,1100,1100],[17.3887036,19.56518781,21.14970947,22.12749193,25.07627933,20.01262831],[63.97315035,64.73629198,63.59157953,66.72374255,66.1197939,66.64414606],[70,71,70,null,70,70],[2.8459109,2.83348788,3.0888032,null,3.1208261,2.8059192],[395.723416,393.9959972,429.4975481,null,433.9503268,390.1625771],[1.2,0.8,0.9,null,0.8,0.8],[4065.587,3990.828,4412.576,571.419,4458.323,4008.456],["S567_AD1","S568_AD2","S569_AD3","S570_AD4","S571_AD5","S572_AD6"],[33.413,31.71,33.549,34.579,32.92,33.082],[5.519,5.907,5.343,5.909,5.966,5.484],[5.503,5.073,5.314,5.262,5.236,5.405],[6.1,6.3,6.3,6.6,6.3,6.1],[0.17,0.19,0.16,0.17,0.18,0.17],[6.35,6.12,6.51,6.22,5.93,6.19],[1.05,1.14,1.04,1.06,1.07,1.03],[1.05,0.98,1.03,0.95,0.9399999999999999,1.01],["S531_AD1","S532_AD2","S533_AD3","S534_AD4","S535_AD5","S536_AD6"],[0,0,0,0,0,0],[0,0,0,0,0,0],[0.7,0.91,1.21,0.68,0.96,0.87],[0.38,1.29,1.89,1.29,1.11,1.95],[32.31,35.21,25.82,65.54000000000001,48.35,61.55],[22.57,17.97,23.43,24.12,28.63,26.12],[0.86,0.8,0.9399999999999999,1.19,1.19,1.35],[0.97,0.8100000000000001,0.99,1.08,1.2,1.06],[1.61,1.54,1.57,1.83,1.77,2.12],[0.27,0.22,0.26,0.32,0.32,0.31],[2.62,2.8,3.09,2.86,2.79,14.3],[0,0,0,0,0,0.15],["#073","#074","#075","#076","#077","#078"],["S561_AD1","S562_AD2","S563_AD3","S564_AD4","S565_AD5","S566_AD6"],[185.33,200.33,212.67,157.67,189.67,201.33],[1.0894,1.1755,1.2229,0.992,1.1757,1.2729],[1.1344,1.1419,1.2266,1.0381,1.2428,1.2273],["15.43","16.30","17.50","12.98","15.66","16.69"],[19.3,20.4,21.9,16.2,19.6,20.9],[514,416,531,474,451,444],[528.17,428.08,546.49,483.24,473.16,454.16],[6.76,6.01,6.9,4.96,9.18,5.46],[525.6900000000001,428.54,545.37,484.01,469.86,453.55],[11.32,10.21,11.51,10.87,10.66,10.46],[4.52,4.24,4.48,4.31,4.14,4.4],[0.97,0.97,0.96,0.97,0.96,0.96],[32.14,29.79,22.24,31.66,25.71,24.36],[87.63,67.98999999999999,91.13,79.48999999999999,74.89,73.48999999999999],[0.72,0.7,0.71,0.7,0.68,0.72],[61.59,60.64,57.99,98.23,85.36,108.91]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>X<\/th>\n      <th>Date<\/th>\n      <th>AD<\/th>\n      <th>SludgeType<\/th>\n      <th>Treatment<\/th>\n      <th>daysop<\/th>\n      <th>week<\/th>\n      <th>Period<\/th>\n      <th>UQ.ID<\/th>\n      <th>SampleID.TSVS<\/th>\n      <th>TS_ww<\/th>\n      <th>VS_wwTS<\/th>\n      <th>VS_ww<\/th>\n      <th>VSr_pct<\/th>\n      <th>SampleID.pH<\/th>\n      <th>pH<\/th>\n      <th>SampleID.CODNH4<\/th>\n      <th>COD_mgL<\/th>\n      <th>TNH3_mgL<\/th>\n      <th>FAN_mgL<\/th>\n      <th>CODr_pct<\/th>\n      <th>CH4_pct<\/th>\n      <th>CH4_L<\/th>\n      <th>CH4_L_kgVS<\/th>\n      <th>O2_pct<\/th>\n      <th>Gasflow_mL<\/th>\n      <th>SampleID.CHN<\/th>\n      <th>C<\/th>\n      <th>H_pct<\/th>\n      <th>N_pct<\/th>\n      <th>CN<\/th>\n      <th>HC<\/th>\n      <th>gC_kg<\/th>\n      <th>gH_kg<\/th>\n      <th>gN_kg<\/th>\n      <th>SampleID.VFA<\/th>\n      <th>Ethanol<\/th>\n      <th>Propanol<\/th>\n      <th>Butanol<\/th>\n      <th>X1.Hexanol<\/th>\n      <th>Acetic.acid<\/th>\n      <th>Propionic.acid<\/th>\n      <th>iso.Butyric.acid<\/th>\n      <th>Butyric.acid<\/th>\n      <th>iso.Valeric.acid<\/th>\n      <th>Valeric.acid<\/th>\n      <th>X4.Methyl.valeric.acid<\/th>\n      <th>Hexanoic.acid<\/th>\n      <th>DNATubeID<\/th>\n      <th>SampleID.DNA<\/th>\n      <th>ngmL_qb<\/th>\n      <th>DNA_mg_gCOD<\/th>\n      <th>DNA_mg_gVS<\/th>\n      <th>DNA_mg_L<\/th>\n      <th>DNA_mgLSludge<\/th>\n      <th>Observed<\/th>\n      <th>Chao1<\/th>\n      <th>se.chao1<\/th>\n      <th>ACE<\/th>\n      <th>se.ACE<\/th>\n      <th>Shannon<\/th>\n      <th>Simpson<\/th>\n      <th>InvSimpson<\/th>\n      <th>Fisher<\/th>\n      <th>pielou_ev<\/th>\n      <th>VFAS<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,6,7,11,12,13,14,16,18,19,20,21,22,23,24,25,26,28,29,30,31,32,33,34,35,37,38,39,40,41,42,43,44,45,46,47,48,51,52,53,55,56,57,58,59,60,61,62,63,64,65,66]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"X","targets":1},{"name":"Date","targets":2},{"name":"AD","targets":3},{"name":"SludgeType","targets":4},{"name":"Treatment","targets":5},{"name":"daysop","targets":6},{"name":"week","targets":7},{"name":"Period","targets":8},{"name":"UQ.ID","targets":9},{"name":"SampleID.TSVS","targets":10},{"name":"TS_ww","targets":11},{"name":"VS_wwTS","targets":12},{"name":"VS_ww","targets":13},{"name":"VSr_pct","targets":14},{"name":"SampleID.pH","targets":15},{"name":"pH","targets":16},{"name":"SampleID.CODNH4","targets":17},{"name":"COD_mgL","targets":18},{"name":"TNH3_mgL","targets":19},{"name":"FAN_mgL","targets":20},{"name":"CODr_pct","targets":21},{"name":"CH4_pct","targets":22},{"name":"CH4_L","targets":23},{"name":"CH4_L_kgVS","targets":24},{"name":"O2_pct","targets":25},{"name":"Gasflow_mL","targets":26},{"name":"SampleID.CHN","targets":27},{"name":"C","targets":28},{"name":"H_pct","targets":29},{"name":"N_pct","targets":30},{"name":"CN","targets":31},{"name":"HC","targets":32},{"name":"gC_kg","targets":33},{"name":"gH_kg","targets":34},{"name":"gN_kg","targets":35},{"name":"SampleID.VFA","targets":36},{"name":"Ethanol","targets":37},{"name":"Propanol","targets":38},{"name":"Butanol","targets":39},{"name":"X1.Hexanol","targets":40},{"name":"Acetic.acid","targets":41},{"name":"Propionic.acid","targets":42},{"name":"iso.Butyric.acid","targets":43},{"name":"Butyric.acid","targets":44},{"name":"iso.Valeric.acid","targets":45},{"name":"Valeric.acid","targets":46},{"name":"X4.Methyl.valeric.acid","targets":47},{"name":"Hexanoic.acid","targets":48},{"name":"DNATubeID","targets":49},{"name":"SampleID.DNA","targets":50},{"name":"ngmL_qb","targets":51},{"name":"DNA_mg_gCOD","targets":52},{"name":"DNA_mg_gVS","targets":53},{"name":"DNA_mg_L","targets":54},{"name":"DNA_mgLSludge","targets":55},{"name":"Observed","targets":56},{"name":"Chao1","targets":57},{"name":"se.chao1","targets":58},{"name":"ACE","targets":59},{"name":"se.ACE","targets":60},{"name":"Shannon","targets":61},{"name":"Simpson","targets":62},{"name":"InvSimpson","targets":63},{"name":"Fisher","targets":64},{"name":"pielou_ev","targets":65},{"name":"VFAS","targets":66}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co"># CREATE A PHYLOSEQ OBJECTS (combined metadata and spectra)</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>psATR <span class="ot">&lt;-</span><span class="fu">phyloseq</span>(</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="fu">otu_table</span>(ATR, <span class="at">taxa_are_rows =</span> T), </span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="fu">sample_data</span>(metadata)</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>)</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>psATRclr <span class="ot">&lt;-</span><span class="fu">phyloseq</span>(</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>  <span class="fu">otu_table</span>(ATRclr, <span class="at">taxa_are_rows =</span> T), </span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>  <span class="fu">sample_data</span>(metadata)</span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>)</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="co"># ABSORBANCE DATA </span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>AVE <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;./data/AVE.csv&quot;</span>)[<span class="sc">-</span><span class="dv">1</span>,] </span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>AVE <span class="ot">&lt;-</span> AVE <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>)<span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">&quot;X&quot;</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>ID) <span class="sc">%&gt;%</span> </span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(S537_AD1<span class="sc">:</span>S1370_AD6, as.numeric))</span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a><span class="co"># OPTION 1: raw</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a>AVE <span class="ot">&lt;-</span>  <span class="fu">as.matrix</span>(AVE)</span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a><span class="co"># OPTION 2: clr transformed</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a>AVEclr <span class="ot">&lt;-</span> compositions<span class="sc">::</span><span class="fu">clr</span>(AVE)</span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a><span class="co"># head(AVEclr)</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a><span class="co"># CREATE A PHYLOSEQ OBJECTS (combined metadata and spectra)</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a>psAVE <span class="ot">&lt;-</span><span class="fu">phyloseq</span>(</span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a>  <span class="fu">otu_table</span>(AVE, <span class="at">taxa_are_rows =</span> T), </span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a>  <span class="fu">sample_data</span>(metadata)</span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a>)</span>
<span id="cb4-28"><a href="#cb4-28" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" tabindex="-1"></a>psAVEclr <span class="ot">&lt;-</span><span class="fu">phyloseq</span>(</span>
<span id="cb4-30"><a href="#cb4-30" tabindex="-1"></a>  <span class="fu">otu_table</span>(AVEclr, <span class="at">taxa_are_rows =</span> T), </span>
<span id="cb4-31"><a href="#cb4-31" tabindex="-1"></a>  <span class="fu">sample_data</span>(metadata)</span>
<span id="cb4-32"><a href="#cb4-32" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="microbial-16s-v3v4-pca" class="section level2 hasAnchor" number="1.2">
<h2 class="hasAnchor"><span class="header-section-number">1.2</span> Microbial 16S (V3V4) PCA<a href="#microbial-16s-v3v4-pca" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Just to see how microbial 16S abundances vary among the selected samples. We expect treated/inhibited samples (15th May) to be different.
<img src="ftir-uq_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
</div>
<div id="raw-spectra" class="section level2 hasAnchor" number="1.3">
<h2 class="hasAnchor"><span class="header-section-number">1.3</span> Raw spectra<a href="#raw-spectra" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="pca-atr" class="section level3 hasAnchor" number="1.3.1">
<h3 class="hasAnchor"><span class="header-section-number">1.3.1</span> PCA ATR<a href="#pca-atr" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="fu">library</span>(compositions)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>my_comparisons <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>symbolsize <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>ps.temp <span class="ot">&lt;-</span> psATR</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">otu_table</span>(ps.temp),  <span class="at">scale =</span> <span class="cn">FALSE</span>, <span class="at">center =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co"># summary(otu_table(ps.temp))</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co"># head(otu_table(ps.temp))</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">get_pca</span>(pca)</span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>cord <span class="ot">&lt;-</span> data<span class="sc">$</span>coord  <span class="co"># extract sample coordinates of PC</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>cor <span class="ot">&lt;-</span> data<span class="sc">$</span>cor</span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a>cos2 <span class="ot">&lt;-</span> data<span class="sc">$</span>cos2</span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a>contrib <span class="ot">&lt;-</span> data<span class="sc">$</span>contrib <span class="co"># contributions of variables</span></span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a><span class="co"># Combine </span></span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a>df.tmp <span class="ot">&lt;-</span> (<span class="fu">data.frame</span>(cord) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>)) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(<span class="fu">sample_data</span>(ps.temp) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>) ) </span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a><span class="co"># compare_means(Dim.1 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.315</span></span>
<span id="cb5-20"><a href="#cb5-20" tabindex="-1"></a><span class="co"># compare_means(Dim.2 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.908</span></span>
<span id="cb5-21"><a href="#cb5-21" tabindex="-1"></a><span class="co"># compare_means(Dim.3 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.0075 **</span></span>
<span id="cb5-22"><a href="#cb5-22" tabindex="-1"></a><span class="co"># compare_means(Dim.4 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.205</span></span></code></pre></div>
<p><img src="ftir-uq_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<div id="compare-variation-of-components" class="section level4 hasAnchor" number="1.3.1.1">
<h4 class="hasAnchor"><span class="header-section-number">1.3.1.1</span> Compare variation of components<a href="#compare-variation-of-components" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Assessing if there were significant differences in the variation of Principal Components (PCs) between groups (control and treatment)
<img src="ftir-uq_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
</div>
</div>
<div id="pca-ave" class="section level3 hasAnchor" number="1.3.2">
<h3 class="hasAnchor"><span class="header-section-number">1.3.2</span> PCA AVE<a href="#pca-ave" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="fu">library</span>(compositions)</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>ps.temp <span class="ot">&lt;-</span> psAVE</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co"># summary(otu_table(ps.temp))</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">data.frame</span>(<span class="fu">otu_table</span>(ps.temp)), <span class="at">scale =</span> <span class="cn">FALSE</span>, <span class="at">center =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co"># otu_table(ps.temp)</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">get_pca</span>(pca)</span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>cord <span class="ot">&lt;-</span> data<span class="sc">$</span>coord</span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>cor <span class="ot">&lt;-</span> data<span class="sc">$</span>cor</span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>cos2 <span class="ot">&lt;-</span> data<span class="sc">$</span>cos2</span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>contrib <span class="ot">&lt;-</span> data<span class="sc">$</span>contrib</span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a>df.tmp <span class="ot">&lt;-</span> (<span class="fu">data.frame</span>(cord) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a>            <span class="fu">left_join</span>(<span class="fu">sample_data</span>(ps.temp) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>) ) </span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a><span class="co"># compare_means(Dim.1 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.133</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a><span class="co"># compare_means(Dim.2 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.397</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a><span class="co"># compare_means(Dim.3 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.026 *</span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a><span class="co"># compare_means(Dim.4 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.502</span></span></code></pre></div>
<p><img src="ftir-uq_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<div id="compare-variation-of-components-1" class="section level4 hasAnchor" number="1.3.2.1">
<h4 class="hasAnchor"><span class="header-section-number">1.3.2.1</span> Compare variation of components<a href="#compare-variation-of-components-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Assessing if there were significant differences in the variation of Principal Components (PCs) between groups (control and treatment)</p>
<p><img src="ftir-uq_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
</div>
</div>
<div id="comparing-atr-and-ave-of-pc3" class="section level3 hasAnchor" number="1.3.3">
<h3 class="hasAnchor"><span class="header-section-number">1.3.3</span> Comparing ATR and AVE of PC3<a href="#comparing-atr-and-ave-of-pc3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="ftir-uq_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
</div>
<div id="spectra-contributions-to-pc3" class="section level3 hasAnchor" number="1.3.4">
<h3 class="hasAnchor"><span class="header-section-number">1.3.4</span> Spectra contributions to PC3<a href="#spectra-contributions-to-pc3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="ftir-uq_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
</div>
</div>
<div id="transformed-spectra---centred-log-ratio" class="section level2 hasAnchor" number="1.4">
<h2 class="hasAnchor"><span class="header-section-number">1.4</span> Transformed spectra - centred-log ratio<a href="#transformed-spectra---centred-log-ratio" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="pca-atr-clr" class="section level3 hasAnchor" number="1.4.1">
<h3 class="hasAnchor"><span class="header-section-number">1.4.1</span> PCA ATR clr<a href="#pca-atr-clr" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="fu">library</span>(compositions)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>ps.temp <span class="ot">&lt;-</span> psATRclr</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">otu_table</span>(ps.temp),  <span class="at">scale =</span> <span class="cn">FALSE</span>, <span class="at">center =</span> <span class="cn">FALSE</span>)</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co"># summary(otu_table(ps.temp))</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="co"># otu_table(ps.temp)</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">get_pca</span>(pca)</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>cord <span class="ot">&lt;-</span> data<span class="sc">$</span>coord</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>cor <span class="ot">&lt;-</span> data<span class="sc">$</span>cor</span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>cos2 <span class="ot">&lt;-</span> data<span class="sc">$</span>cos2</span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>contrib <span class="ot">&lt;-</span> data<span class="sc">$</span>contrib</span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>df.tmp <span class="ot">&lt;-</span> (<span class="fu">data.frame</span>(cord) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>)) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(<span class="fu">sample_data</span>(ps.temp) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>) ) </span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a><span class="co"># compare_means(Dim.1 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.281</span></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a><span class="co"># compare_means(Dim.2 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.627</span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a><span class="co"># compare_means(Dim.3 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.00117 **</span></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a><span class="co"># compare_means(Dim.4 ~ Treatment, method = &quot;t.test&quot;, df.tmp) # 0.518</span></span></code></pre></div>
<p><img src="ftir-uq_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<div id="compare-variation-of-components-2" class="section level4 hasAnchor" number="1.4.1.1">
<h4 class="hasAnchor"><span class="header-section-number">1.4.1.1</span> Compare variation of components<a href="#compare-variation-of-components-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Assessing if there were significant differences in the variation of Principal Components (PCs) between groups (control and treatment)</p>
<p><img src="ftir-uq_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
</div>
</div>
<div id="pca-ave---clr" class="section level3 hasAnchor" number="1.4.2">
<h3 class="hasAnchor"><span class="header-section-number">1.4.2</span> PCA AVE - clr<a href="#pca-ave---clr" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="fu">library</span>(compositions)</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>my_comparisons <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>ps.temp <span class="ot">&lt;-</span> psAVEclr</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co"># summary(otu_table(ps.temp))</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">data.frame</span>(<span class="fu">otu_table</span>(ps.temp)), <span class="at">scale =</span> <span class="cn">FALSE</span>, <span class="at">center =</span> <span class="cn">FALSE</span>)  <span class="co"># FALSE FOR BOTH - CLR data should already be centred and scaled. </span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="co"># otu_table(ps.temp)</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">get_pca</span>(pca)</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>cord <span class="ot">&lt;-</span> data<span class="sc">$</span>coord</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>cor <span class="ot">&lt;-</span> data<span class="sc">$</span>cor</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>cos2 <span class="ot">&lt;-</span> data<span class="sc">$</span>cos2</span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>contrib <span class="ot">&lt;-</span> data<span class="sc">$</span>contrib</span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>df.tmp <span class="ot">&lt;-</span> (<span class="fu">data.frame</span>(cord) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>)) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(<span class="fu">sample_data</span>(ps.temp) <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>) ) </span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a><span class="co"># compare_means(Dim.1 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.172</span></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a><span class="co"># compare_means(Dim.2 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.708</span></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a><span class="co"># compare_means(Dim.3 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.0228 *</span></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a><span class="co"># compare_means(Dim.4 ~ Treatment, method = &quot;t.test&quot;, df.tmp) #0.613</span></span></code></pre></div>
<p><img src="ftir-uq_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<div id="compare-variation-of-components-3" class="section level4 hasAnchor" number="1.4.2.1">
<h4 class="hasAnchor"><span class="header-section-number">1.4.2.1</span> Compare variation of components<a href="#compare-variation-of-components-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Assessing if there were significant differences in the variation of Principal Components (PCs) between groups (control and treatment)</p>
<p><img src="ftir-uq_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
</div>
<div id="comparing-atr-and-ave-of-pc3-1" class="section level3 hasAnchor" number="1.4.3">
<h3 class="hasAnchor"><span class="header-section-number">1.4.3</span> Comparing ATR and AVE of PC3<a href="#comparing-atr-and-ave-of-pc3-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="ftir-uq_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
</div>
<div id="spectra-contributions-to-pc3-1" class="section level3 hasAnchor" number="1.4.4">
<h3 class="hasAnchor"><span class="header-section-number">1.4.4</span> Spectra contributions to PC3<a href="#spectra-contributions-to-pc3-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Unsure why the clr-transformed ATR spectra resulted in such different PC3 contributions compared to clr-transformed AVE spectra.</p>
<p><img src="ftir-uq_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
</div>
</div>
<div id="corplot" class="section level2 hasAnchor" number="1.5">
<h2 class="hasAnchor"><span class="header-section-number">1.5</span> Corplot<a href="#corplot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This needs further assessments/regression etc as it was done without any consideration to data distribution / validity.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">library</span>(ggcorrplot)</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="co"># Compute a correlation matrix</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>df.cor <span class="ot">&lt;-</span> df.tmp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ID <span class="sc">!=</span> <span class="st">&quot;S963_AD6&quot;</span>) <span class="sc">%&gt;%</span>  <span class="co"># filter this sample as it has NAs</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Dim<span class="fl">.1</span>, Dim<span class="fl">.2</span>, Dim<span class="fl">.3</span>, Dim<span class="fl">.4</span>, Dim<span class="fl">.5</span>, Gasflow_mL, C, H_pct, N_pct, CN, HC, Ethanol<span class="sc">:</span>Hexanoic.acid, ngmL_qb, DNA_mg_gCOD)</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>corr <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">cor</span>(df.cor, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>), <span class="dv">1</span>) </span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggcorrplot</span>(corr)</span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>p <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;AVE clr correlations to variables&quot;</span>)</span></code></pre></div>
<p><img src="ftir-uq_files/figure-html/unnamed-chunk-24-1.png" width="960" /></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="co">#ggsave(plot = p, &quot;./Figures/PC3correlations_clr.png&quot;, height=18, width=18, units=&#39;cm&#39;, dpi=300)</span></span></code></pre></div>
<!--chapter:end:index.Rmd-->
</div>
</div>
<!--bookdown:body:end-->
            </section>

          </div>
        </div>
      </div>
<!--bookdown:link_prev-->
<!--bookdown:link_next-->
    </div>
  </div>
<!--bookdown:config-->

</body>

</html>
